docs/gallery/plot_external_resources.py

```diff
@@ -320,3 +323,33 @@ def __init__(self, **kwargs):
 
 er_read = ExternalResources.from_norm_tsv(path='./')
 os.remove('./er.zip')
+
+###############################################################################
+# Using TermSet with ExternalResources
+# ------------------------------------------------
+# :py:class:`~hdmf.TermSet` allows for an easier way to add references to
+# :py:class:`~hdmf.common.resources.ExternalResources`. These enumerations take place of the
+# entity_id and entity_uri parameters. :py:class:`~hdmf.common.resources.Key` values will have
+# to match the name of the term in the :py:class:`~hdmf.TermSet`.
+try:
+    import linkml_runtime  # noqa: F401
+    LINKML_INSTALLED = True
+except ImportError:
+    LINKML_INSTALLED = False
+
+if LINKML_INSTALLED:
+    from hdmf.term_set import TermSet
+
+    terms = TermSet(name='Species', term_schema_path='docs/gallery/example_term_set.yaml')
+    col1 = VectorData(
+        name='Species_Data',
+        description='...',
+        data=['Homo sapiens', 'Ursus arctos horribilis'],
+        term_set=terms,
+    )
+
+    species = DynamicTable(name='species', description='My species', columns=[col1],)
+    er.add_ref_term_set(file=file,
+                        container=species,
+                        attribute='Species_Data',
+                       )
```

Comment:
Maybe this is part of PyNWB, but do we have a function to call `add_ref_term_set` for all objects that have term_set linked to it? E.g., could I call some version of `add_ref_term_set` with the `species` DynamicTable object and it would figure out all the columns in the table that have a term set and add them to the ExternalResources?